www/library.html

```diff
@@ -0,0 +1,15 @@
+<!DOCTYPE html>
+<html>
+  <head>
+    <meta charset="utf-8" />
+    <title>Placeholder for injecting an article into the iframe</title>
+    <meta
+      http-equiv="Content-Security-Policy"
+      content="default-src 'self' data: * 'unsafe-inline' 'unsafe-eval'; frame-src 'self' moz-extension: chrome-extension: *; object-src 'none';"
+    />
+    <script src="./js/library.js" ></script>
```

Comment:
It seems like when we build the actual app all the files are moved from `js/lib/*` to `js/*` which seems to cause a problem where when we `npm run preview` I get a 404 since there is no file at `js/library.js`

Could you tell me how to change this path only during compile time from `js/lib/library.js` -> `js/library.js`

I have already tried `replace()` from `rollup.config.js` but can't make it work @Jaifroid

Reply:
All JS files should be bundled into bundle.js in rollup.config.js. I think you only need to add an import statement at the top of a file that uses it.

Reply:
Ah sorry, I didn't look at the code, and I think this script file is only used inside library.html. In that case, you don't need to (can't) turn it into a module or bundle it, you can instead put it in the same folder as library.html and ensure both are copied by rollup. In fact if both are in www, I think they should be copied automatically, and you don't need to touch the rollup config.

Reply:
If you don't like putting it in the same folder (i.e. in `www`, not in a subfolder), then you would have to do a search-and-replace inside the rollup config, which would search for the path `./js/lib/library.js` and replace it to `./js/library.js`, when you copy `library.html`. There are a few examples of doing such a search-and-replace in rollup config (a simple javascript `.replace` function can do it).

Note that you have to do it in two places, in the PRODUCTION (minified) buiild, and also in the normal (unminified) build. There are two code blocks in rollup config, one for production and the other for normal. There is a lot of duplicated code in the rollup config, which I'm not at all proud of, but it kind of built up like this from a small amount, and then mushroomed. I aim to find a way to share repeated code when I have a chance.